### PR TITLE
Use double spaces instead of Chatterino character when possible

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

No issue. Can help eliminate issues with users noticing the Chatterino character. Double spaces are way less noticeable (Chatterino, Chatty and webchat don't even render those!) than a space and U+E0000.

### CODE REVIEW

**Comment by zneix** [comment]
> when possible

Am I missing soemthing or are you straight-up removing our old solution instead?

**Comment by Mm2PL** [comment]
If there is no space, there is nothing to replace with a double space.

**Review by pajlada — APPROVED** [review]

File: src/providers/twitch/TwitchChannel.cpp [hunk 1]
```diff
@@ -375,7 +375,16 @@ void TwitchChannel::sendMessage(const QString &message)
...
             {
-                parsedMessage.append(MAGIC_MESSAGE_SUFFIX);
+                auto spaceIndex = parsedMessage.indexOf(' ');
+                if (spaceIndex == -1)
+                {
+                    parsedMessage.append(MAGIC_MESSAGE_SUFFIX);
+                }
+                else
+                {
+                    // replace only first space with a double space
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchChannel.cpp:385]:
```suggestion
                    // replace the space we found in spaceIndex with two spaces
```

File: src/providers/twitch/TwitchChannel.cpp [hunk 2]
```diff
@@ -375,7 +375,16 @@ void TwitchChannel::sendMessage(const QString &message)
         {
             if (parsedMessage == this->lastSentMessage_)
             {
-                parsedMessage.append(MAGIC_MESSAGE_SUFFIX);
+                auto spaceIndex = parsedMessage.indexOf(' ');
+                if (spaceIndex == -1)
+                {
+                    parsedMessage.append(MAGIC_MESSAGE_SUFFIX);
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchChannel.cpp:381]:
```suggestion
                    // no spaces found, fall back to old magic character
                    parsedMessage.append(MAGIC_MESSAGE_SUFFIX);
```

**Review by zneix — APPROVED** [review]

**Comment by nilyt** [comment]
I didn't even know this was a feature, until I saw this change.
Always considered this behaviour a bug I wanted to fix eventually, as it happens seemingly randomly, breaks building SE pyramids, and was just a nuisance all around (to various people).

Where is this setting (`getSettings()->allowDuplicateMessages`) in the UI, so I can turn it off? Or is it not exposed in the UI (for reasons)?
I'd vote for it to be off by default for the mentioned reasons. At least I can't recall enabling something like that?
Or maybe I did? The name "allowDuplicateMessages" is something I'd associate with 'r9k' more than with what it does.

**Comment by Mm2PL** [comment]
> Where is this setting (getSettings()->allowDuplicateMessages) in the UI, so I can turn it off?

here

**Pull request merged**